lib/ansible/module_utils/docker_common.py

```diff
@@ -410,12 +411,20 @@ def _image_lookup(self, name, tag):
         exists.
         '''
         try:
-            response = self.images(name=name)
+            if name in DOCKER_RESERVED_IMAGE_NAMES:
+                response = self.images()
+            else:
+                response = self.images(name=name)
         except Exception as exc:
             self.fail("Error searching for image %s - %s" % (name, str(exc)))
         images = response
-        if tag:
-            lookup = "%s:%s" % (name, tag)
+        lookup = "%s:%s" % (name, tag)
+        if tag and name in DOCKER_RESERVED_IMAGE_NAMES:
+            for image in response:
+                if image['Id'] == lookup:
```

Comment:
Hm interesting thought. Docker command line does allow short form IDs - ala on my machine `docker inspect 8fc` works because it's unique to one image currently on my system.

This currently does not work with the code in this module (complains "registry returned more then 1 result") and I'm inclined to say that for an ansible module you should specify an exact hash if you're going this path, but the behaviour should definitely be defined.